### PR TITLE
refactor(kubeadm): remove deprecated command "alpha selfhosting pivot"

### PR DESCRIPTION
**Pull request by knight42**
**What type of PR is this?**

/kind deprecation
/kind cleanup

**What this PR does / why we need it**:

Remove deprecated command "kubeadm alpha selfhosting pivot".

**Which issue(s) this PR fixes**:

xref: https://github.com/kubernetes/kubeadm/issues/2299

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
kubeadm: deprecated command "alpha selfhosting pivot" is removed now.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cmd/kubeadm/app/cmd/alpha/selfhosting.go [hunk 1]
```diff
@@ -1,161 +0,0 @@
...
-
-			// This call returns the ready-to-use configuration based on the configuration file that might or might not exist and the default cfg populated by flags
-			internalcfg, err := configutil.LoadOrDefaultInitConfiguration(cfgPath, &kubeadmapiv1beta2.InitConfiguration{}, cfg)
-			if err != nil {
-				return err
-			}
-
-			// Converts the Static Pod-hosted control plane into a self-hosted one
-			waiter := apiclient.NewKubeWaiter(client, 2*time.Minute, os.Stdout)
-			return selfhosting.CreateSelfHostedControlPlane(constants.GetStaticPodDirectory(), constants.KubernetesDir, internalcfg, client, waiter, false, certsInSecrets)
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/alpha/selfhosting.go:137]: please remove the selfhosting package under phases too:
`k8s.io/kubernetes/cmd/kubeadm/app/phases/selfhosting`

Comment (knight42) [reply cmd/kubeadm/app/cmd/alpha/selfhosting.go:137]: fixed

**Comment by neolit123** [comment]
looks like there are more things to clean up under the upgrade and constants packages:

```
controlplane $ grep "selfhost" * -rniI
app/cmd/alpha/alpha.go:38:      shCommand := newCmdSelfhosting(in)
app/cmd/alpha/BUILD:9:        "selfhosting.go",
app/cmd/alpha/BUILD:27:        "//cmd/kubeadm/app/phases/selfhosting:go_default_library",
app/cmd/alpha/selfhosting.go:38:        "k8s.io/kubernetes/cmd/kubeadm/app/phases/selfhosting"
app/cmd/alpha/selfhosting.go:45:        selfhostingLongDesc = cmdutil.LongDesc(`
app/cmd/alpha/selfhosting.go:52:        selfhostingExample = cmdutil.Examples(`
app/cmd/alpha/selfhosting.go:59:// newCmdSelfhosting returns the self-hosting Cobra command
app/cmd/alpha/selfhosting.go:60:func newCmdSelfhosting(in io.Reader) *cobra.Command {
app/cmd/alpha/selfhosting.go:62:                Use:     "selfhosting",
app/cmd/alpha/selfhosting.go:63:                Aliases: []string{"selfhosted", "self-hosting"},
app/cmd/alpha/selfhosting.go:68:        cmd.AddCommand(getSelfhostingSubCommand(in))
app/cmd/alpha/selfhosting.go:72:// getSelfhostingSubCommand returns sub commands for Self-hosting phase
app/cmd/alpha/selfhosting.go:73:func getSelfhostingSubCommand(in io.Reader) *cobra.Command {
app/cmd/alpha/selfhosting.go:87:                Long:    selfhostingLongDesc,
app/cmd/alpha/selfhosting.go:88:                Example: selfhostingExample,
app/cmd/alpha/selfhosting.go:137:                       return selfhosting.CreateSelfHostedControlPlane(constants.GetStaticPodDirectory(), constants.KubernetesDir, internalcfg, client, waiter, false, certsInSecrets)
app/cmd/upgrade/common.go:131:  if upgrade.IsControlPlaneSelfHosted(client) {
app/constants/constants_test.go:113:func TestAddSelfHostedPrefix(t *testing.T) {
app/constants/constants_test.go:136:                    actual := AddSelfHostedPrefix(rt.componentName)
app/constants/constants_test.go:139:                                    "failed AddSelfHostedPrefix:\n\texpected: %s\n\t  actual: %s",
app/constants/constants.go:307: // SelfHostingPrefix describes the prefix workloads that are self-hosted by kubeadm has
app/constants/constants.go:308: SelfHostingPrefix = "self-hosted-"
app/constants/constants.go:558:// AddSelfHostedPrefix adds the self-hosted- prefix to the component name
app/constants/constants.go:559:func AddSelfHostedPrefix(componentName string) string {
app/constants/constants.go:560: return fmt.Sprintf("%s%s", SelfHostingPrefix, componentName)
app/phases/selfhosting/selfhosting_test.go:17:package selfhosting
app/phases/selfhosting/selfhosting_volumes_test.go:17:package selfhosting
app/phases/selfhosting/selfhosting_volumes.go:17:package selfhosting
app/phases/selfhosting/podspec_mutation_test.go:17:package selfhosting
app/phases/selfhosting/podspec_mutation_test.go:290:func TestSetSelfHostedVolumesForAPIServer(t *testing.T) {
app/phases/selfhosting/podspec_mutation_test.go:298:                    name: "set selfhosted volumes for api server",
app/phases/selfhosting/podspec_mutation_test.go:377:                    setSelfHostedVolumesForAPIServer(rt.podSpec)
app/phases/selfhosting/podspec_mutation_test.go:382:                            t.Errorf("failed setSelfHostedVolumesForAPIServer:\nexpected:\n%v\nsaw:\n%v", rt.expected, *rt.podSpec)
app/phases/selfhosting/podspec_mutation_test.go:388:func TestSetSelfHostedVolumesForControllerManager(t *testing.T) {
app/phases/selfhosting/podspec_mutation_test.go:397:                    name: "set selfhosted volumes for controller mananger",
app/phases/selfhosting/podspec_mutation_test.go:503:                    setSelfHostedVolumesForControllerManager(rt.podSpec)
app/phases/selfhosting/podspec_mutation_test.go:508:                            t.Errorf("failed setSelfHostedVolumesForControllerManager:\nexpected:\n%v\nsaw:\n%v", rt.expected, *rt.podSpec)
app/phases/selfhosting/podspec_mutation_test.go:514:func TestSetSelfHostedVolumesForScheduler(t *testing.T) {
app/phases/selfhosting/podspec_mutation_test.go:522:                    name: "set selfhosted volumes for scheduler",
app/phases/selfhosting/podspec_mutation_test.go:581:                    setSelfHostedVolumesForScheduler(rt.podSpec)
app/phases/selfhosting/podspec_mutation_test.go:586:                            t.Errorf("failed setSelfHostedVolumesForScheduler:\nexpected:\n%v\nsaw:\n%v", rt.expected, *rt.podSpec)
app/phases/selfhosting/podspec_mutation.go:17:package selfhosting
app/phases/selfhosting/podspec_mutation.go:29:  // selfHostedKubeConfigDir sets the directory where kubeconfig files for the scheduler and controller-manager should be mounted
app/phases/selfhosting/podspec_mutation.go:32:  selfHostedKubeConfigDir = "/etc/kubernetes/kubeconfig"
app/phases/selfhosting/podspec_mutation.go:68:          mutators[kubeadmconstants.KubeAPIServer] = append(mutators[kubeadmconstants.KubeAPIServer], setSelfHostedVolumesForAPIServer)
app/phases/selfhosting/podspec_mutation.go:69:          mutators[kubeadmconstants.KubeControllerManager] = append(mutators[kubeadmconstants.KubeControllerManager], setSelfHostedVolumesForControllerManager)
app/phases/selfhosting/podspec_mutation.go:70:          mutators[kubeadmconstants.KubeScheduler] = append(mutators[kubeadmconstants.KubeScheduler], setSelfHostedVolumesForScheduler)
app/phases/selfhosting/podspec_mutation.go:131:// setSelfHostedVolumesForAPIServer makes sure the self-hosted api server has the right volume source coming from a self-hosted cluster
app/phases/selfhosting/podspec_mutation.go:132:func setSelfHostedVolumesForAPIServer(podSpec *v1.PodSpec) {
app/phases/selfhosting/podspec_mutation.go:141:// setSelfHostedVolumesForControllerManager makes sure the self-hosted controller manager has the right volume source coming from a self-hosted cluster
app/phases/selfhosting/podspec_mutation.go:142:func setSelfHostedVolumesForControllerManager(podSpec *v1.PodSpec) {
app/phases/selfhosting/podspec_mutation.go:152: // Change directory for the kubeconfig directory to selfHostedKubeConfigDir
app/phases/selfhosting/podspec_mutation.go:155:                 podSpec.Containers[0].VolumeMounts[i].MountPath = selfHostedKubeConfigDir
app/phases/selfhosting/podspec_mutation.go:163:         controllerManagerKubeConfigPath := filepath.Join(selfHostedKubeConfigDir, kubeadmconstants.ControllerManagerKubeConfigFileName)
app/phases/selfhosting/podspec_mutation.go:175:// setSelfHostedVolumesForScheduler makes sure the self-hosted scheduler has theright volume source coming from a self-hosted cluster
app/phases/selfhosting/podspec_mutation.go:176:func setSelfHostedVolumesForScheduler(podSpec *v1.PodSpec) {
app/phases/selfhosting/podspec_mutation.go:184: // Change directory for the kubeconfig directory to selfHostedKubeConfigDir
app/phases/selfhosting/podspec_mutation.go:187:                 podSpec.Containers[0].VolumeMounts[i].MountPath = selfHostedKubeConfigDir
app/phases/selfhosting/podspec_mutation.go:195:         schedulerKubeConfigPath := filepath.Join(selfHostedKubeConfigDir, kubeadmconstants.SchedulerKubeConfigFileName)
app/phases/selfhosting/BUILD:13:        "selfhosting_test.go",
app/phases/selfhosting/BUILD:14:        "selfhosting_volumes_test.go",
app/phases/selfhosting/BUILD:30:        "selfhosting.go",
app/phases/selfhosting/BUILD:31:        "selfhosting_volumes.go",
app/phases/selfhosting/BUILD:33:    importpath = "k8s.io/kubernetes/cmd/kubeadm/app/phases/selfhosting",
app/phases/selfhosting/selfhosting.go:17:package selfhosting
app/phases/selfhosting/selfhosting.go:40:       // selfHostingWaitTimeout describes the maximum amount of time a self-hosting wait process should wait before timing out
app/phases/selfhosting/selfhosting.go:41:       selfHostingWaitTimeout = 2 * time.Minute
app/phases/selfhosting/selfhosting.go:43:       // selfHostingFailureThreshold describes how many times kubeadm will retry creating the DaemonSets
app/phases/selfhosting/selfhosting.go:44:       selfHostingFailureThreshold int = 5
app/phases/selfhosting/selfhosting.go:47:// CreateSelfHostedControlPlane is responsible for turning a Static Pod-hosted controlplane to a self-hosted one
app/phases/selfhosting/selfhosting.go:59:func CreateSelfHostedControlPlane(manifestsDir, kubeConfigDir string, cfg *kubeadmapi.InitConfiguration, client clientset.Interface, waiter apiclient.Waiter, dryRun bool, certsInSecrets bool) error {
app/phases/selfhosting/selfhosting.go:62:       waiter.SetTimeout(selfHostingWaitTimeout)
app/phases/selfhosting/selfhosting.go:97:               // Create or update the DaemonSet in the API Server, and retry selfHostingFailureThreshold times if it errors out
app/phases/selfhosting/selfhosting.go:100:              }, selfHostingFailureThreshold); err != nil {
app/phases/selfhosting/selfhosting.go:105:              if err := waiter.WaitForPodsWithLabel(BuildSelfHostedComponentLabelQuery(componentName)); err != nil {
app/phases/selfhosting/selfhosting.go:143:                      Name:      kubeadmconstants.AddSelfHostedPrefix(name),
app/phases/selfhosting/selfhosting.go:145:                      Labels:    BuildSelfhostedComponentLabels(name),
app/phases/selfhosting/selfhosting.go:149:                              MatchLabels: BuildSelfhostedComponentLabels(name),
app/phases/selfhosting/selfhosting.go:153:                                      Labels: BuildSelfhostedComponentLabels(name),
app/phases/selfhosting/selfhosting.go:165:// BuildSelfhostedComponentLabels returns the labels for a self-hosted component
app/phases/selfhosting/selfhosting.go:166:func BuildSelfhostedComponentLabels(component string) map[string]string {
app/phases/selfhosting/selfhosting.go:168:              "k8s-app": kubeadmconstants.AddSelfHostedPrefix(component),
app/phases/selfhosting/selfhosting.go:172:// BuildSelfHostedComponentLabelQuery creates the right query for matching a self-hosted Pod
app/phases/selfhosting/selfhosting.go:173:func BuildSelfHostedComponentLabelQuery(componentName string) string {
app/phases/selfhosting/selfhosting.go:174:      return fmt.Sprintf("k8s-app=%s", kubeadmconstants.AddSelfHostedPrefix(componentName))
app/phases/upgrade/health.go:260:// IsControlPlaneSelfHosted returns whether the control plane is self hosted or not
app/phases/upgrade/health.go:261:func IsControlPlaneSelfHosted(client clientset.Interface) bool {
app/phases/upgrade/health.go:267:       // If there are no NotReady DaemonSets, we are using selfhosting
app/phases/upgrade/health.go:275:               dsName := constants.AddSelfHostedPrefix(component)
app/BUILD:51:        "//cmd/kubeadm/app/phases/selfhosting:all-srcs",
```

**Comment by knight42** [comment]
@neolit123 I think I have cleaned all of them now. PTAL

**Review by neolit123 — COMMENTED** [review]
thanks
/lgtm
/approve
/priority important-longterm

**Review by ChiefyG-Admin — COMMENTED** [review]
Which part?

**Pull request merged**